precompiles/avs/tx.go

```diff
@@ -305,5 +326,108 @@ func (p Precompile) RegisterBLSPublicKey(
 		return nil, err
 	}
 
+	if err = p.EmitPublicKeyRegistered(ctx, stateDB, blsParams); err != nil {
+		return nil, err
+	}
+	return method.Outputs.Pack(true)
+}
+
+// RegisterOperatorToExocore operator register.
+func (p Precompile) RegisterOperatorToExocore(
```

Comment:
Is it necessary to add this function here? Operators should ideally register using the Cosmos transaction type only and not through this method.

Even if this function is added here, the commission, the earnings address, the approve address and the client chain address members of the `OperatorInfo` struct are not set or validated in this call.

CC @TimmyExogenous

Reply:
fixed